### PR TITLE
[Auto PLUGIN] update Auto docs

### PR DESCRIPTION
**Pull request by tiger100256-hu**
Signed-off-by: Hu, Yuan2 <yuan2.hu@intel.com>

### Details:
 - *continue to fix  the  review in [10623](https://github.com/openvinotoolkit/openvino/pull/10623)*

### Tickets:
 - *80063*

### CODE REVIEW

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 1]
```diff
@@ -74,27 +74,27 @@ Inference with AUTO is configured similarly to other plugins: first you configur
...
+|                           | | comma-separated, no spaces                  | | The device sequence will be taken as priority           |
+|                           | |                                             | | from high to low.                                       |
+|                           | |                                             | | This configuration is optional.                         |
++---------------------------+-----------------------------------------------+-----------------------------------------------------------+
+| ov::hint::performance_mode| | ov::hint::PerformanceMode::LATENCY          | | Specifies the performance mode preferred                |
+|                           | | ov::hint::PerformanceMode::THROUGHPUT       | | by the application.                                     |
++---------------------------+-----------------------------------------------+-----------------------------------------------------------+
+| ov::hint::model_priority  | | ov::hint::MEDIUM                            | | Indicates the priority for a network.                   |
```
Comment (peterchen-intel) [inline docs/OV_Runtime_UG/auto_device_selection.md:94]:
```suggestion
| ov::hint::model_priority  | | ov::hint::Priority::MEDIUM                  | | Indicates the priority for a network.                   |
```

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 2]
```diff
@@ -74,27 +74,27 @@ Inference with AUTO is configured similarly to other plugins: first you configur
...
+|                           | | ov::hint::HIGH                              | | Importantly!                                            |
```
Comment (peterchen-intel) [inline docs/OV_Runtime_UG/auto_device_selection.md:95]:
```suggestion
|                           | | ov::hint::Priority::HIGH                    | | Importantly!                                            |
```

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 3]
```diff
@@ -74,27 +74,27 @@ Inference with AUTO is configured similarly to other plugins: first you configur
...
+|                           | | ov::hint::LOW                               | | This property is still not fully supported              |
```
Comment (peterchen-intel) [inline docs/OV_Runtime_UG/auto_device_selection.md:96]:
```suggestion
|                           | | ov::hint::Priority::LOW                     | | This property is still not fully supported              |
```

**Review by peterchen-intel — COMMENTED** [review]
1. Remove VPUX related.
2. Intel® Neural Compute Stick 2 (Intel® NCS2) ->
Intel® Movidius™ Myriad™ X VPU
(e.g. Intel® Neural Compute Stick 2 (Intel® NCS2))

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 1 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md]: fix it

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 2 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md]: fix it

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 3 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md]: fix it

**Comment by tiger100256-hu** [comment]
removed VPUX and update the MYRAID full name

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 4]
```diff
@@ -119,32 +116,31 @@ The following commands are accepted by the API:
 
    .. code-block:: cpp
 
-      /*** With Inference Engine 2.0 API ***/
+      /*** With OpenVINO Runtime API 2.0 ***/
       ov::Core core; 
 
       // Read a network in IR, PaddlePaddle, or ONNX format:
-      std::shared_ptr<ov::Model> model = core.read_model("sample.xml");    
+      std::shared_ptr<ov::Model> model = core.read_model("sample.xml");
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:123]: move these snippets to .cpp file

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 5]
```diff
@@ -153,66 +149,66 @@ The following commands are accepted by the API:
...
-      exec_net = ie.load_network(network=net, device_name="AUTO", config={"MULTI_DEVICE_PRIORITIES": "CPU,GPU"})
-      
+      exec_net = ie.load_network(network=net, device_name="AUTO:GPU,CPU")
+      exec_net = ie.load_network(network=net, device_name="AUTO", config={"MULTI_DEVICE_PRIORITIES": "GPU,CPU"})
+
       # the AUTO plugin is pre-configured (globally) with the explicit option:
-      ie.SetConfig(config={"MULTI_DEVICE_PRIORITIES", "CPU,GPU"}, device_name="AUTO");
+      ie.set_config(config={"MULTI_DEVICE_PRIORITIES":"GPU,CPU"}, device_name="AUTO");
 
 @endsphinxdirective
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:213]: let's refer to openvino_docs_OV_UG_query_api.html#get-a-set-of-available-devices in the text below instead of writing manual snippets

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 6]
```diff
@@ -290,93 +288,92 @@ The property enables you to control the priorities of networks in the Auto-Devic
...
       // Read a network in IR, PaddlePaddle, or ONNX format
-      stdshared_ptrovModel model = core.read_model(sample.xml);
+      std::shared_ptr<ov::Model> model = core.read_model("sample.xml");
 
-      // Configure the VPUX and the Myriad devices separately and load the network to the Auto-Device plugin
-      set VPU config
-      core.set_property(VPUX, {});
+      // Configure the CPU and the Myriad devices separately and load the network to the Auto-Device plugin
+      // set CPU config
+      core.set_property("CPU", {});
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:355]: we can also use `ov::device_properties` (see openvino_docs_OV_UG_Hetero_execution.html#configure-fallback-devices)

**Review by ilya-lavrenov — COMMENTED** [review]
move all snippets to .cpp and .py files

File: docs/snippets/AUTO0.cpp [hunk 7]
```diff
@@ -1,12 +1,27 @@
...
 int main() {
 //! [part0]
-    InferenceEngine::Core ie;
-    InferenceEngine::CNNNetwork network = ie.ReadNetwork("sample.xml");
-    // these 2 lines below are equivalent
-    InferenceEngine::ExecutableNetwork exec0 = ie.LoadNetwork(network, "AUTO");
-    InferenceEngine::ExecutableNetwork exec1 = ie.LoadNetwork(network, "");
+	/*********************
+	 * With OpenVINO Runtime API 2.0
+	 *********************/
```
[the remark this answers is not in the document]
Comment (tiger100256-hu) [reply docs/snippets/AUTO0.cpp]: fixed it

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 8]
```diff
@@ -13,30 +13,27 @@
 The Auto-Device plugin, or AUTO, is a virtual device which automatically selects the processing unit to use for inference with OpenVINO™. It chooses from a list of available devices defined by the user and aims at finding the most suitable hardware for the given model. The best device is chosen using the following logic: 
 
 1. Check which supported devices are available. 
-2. Check the precision of the input model (for detailed information on precisions read more on the [OPTIMIZATION_CAPABILITIES metric](../IE_PLUGIN_DG/Plugin.md)) 
+2. Check the precision of the input model (for detailed information on precisions read more on the `ov::device::capabilities`)
```
[the remark this answers is not in the document]
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md:16]: @wangleis @kblaszczak-intel please help to have a look

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 9]
```diff
@@ -13,30 +13,27 @@
 The Auto-Device plugin, or AUTO, is a virtual device which automatically selects the processing unit to use for inference with OpenVINO™. It chooses from a list of available devices defined by the user and aims at finding the most suitable hardware for the given model. The best device is chosen using the following logic: 
 
 1. Check which supported devices are available. 
-2. Check the precision of the input model (for detailed information on precisions read more on the [OPTIMIZATION_CAPABILITIES metric](../IE_PLUGIN_DG/Plugin.md)) 
+2. Check the precision of the input model (for detailed information on precisions read more on the `ov::device::capabilities`) 
 3. From the priority list, select the first device capable of supporting the given precision.
```
[the remark this answers is not in the document]
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md:17]: @wangleis @kblaszczak-intel please help to have a look

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 10]
```diff
@@ -74,27 +71,27 @@ Inference with AUTO is configured similarly to other plugins: first you configur
...
+| ov::hint::performance_mode| | ov::hint::PerformanceMode::LATENCY          | | Specifies the performance mode preferred                |
+|                           | | ov::hint::PerformanceMode::THROUGHPUT       | | by the application.                                     |
++---------------------------+-----------------------------------------------+-----------------------------------------------------------+
+| ov::hint::model_priority  | | ov::hint::Priority::HIGH                    | | Indicates the priority for a network.                   |
+|                           | | ov::hint::Priority::MEDIUM                  | | Importantly!                                            |
+|                           | | ov::hint::Priority::LOW                     | | This property is still not fully supported              |
++---------------------------+-----------------------------------------------+-----------------------------------------------------------+
 @endsphinxdirective
 
 @sphinxdirective
```
[the remark this answers is not in the document]
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md:97]: remove  the line contain CommonTestUtils::DEVICE_GPU
@wangleis @kblaszczak-intel please help to have a look

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 11]
```diff
@@ -119,32 +116,31 @@ The following commands are accepted by the API:
 
    .. code-block:: cpp
 
-      /*** With Inference Engine 2.0 API ***/
+      /*** With OpenVINO Runtime API 2.0 ***/
```
[the remark this answers is not in the document]
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md]: fixed it

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 4 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md:121]: @kblaszczak-intel  do you agree to move snippets to .cpp and .py ?

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 5 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md:207]: @ilya-lavrenov  I didn't get the link, could you shared it again

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 6 above)
Comment (tiger100256-hu) [reply docs/OV_Runtime_UG/auto_device_selection.md]: fixed it

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (wangleis) [reply docs/OV_Runtime_UG/auto_device_selection.md:16]: @ilya-lavrenov how about change "precision of the input model" to "precision of the input network" or "precision of the input graph"?

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (wangleis) [reply docs/OV_Runtime_UG/auto_device_selection.md:17]: @ilya-lavrenov Yes, you are right. When priority list is empty, AUTO will create priority list include all devices based on internal static logic as default.

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (wangleis) [reply docs/OV_Runtime_UG/auto_device_selection.md:97]: @ilya-lavrenov Updated. Please review again. Thanks.

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/auto_device_selection.md:17]: As we are talking from the app's perspective, I think the priority list is whatever devices are available, so by default it is all of them, like in the table below. To make it more informative and clear, we could make it:

From the priority list, including all available devices if the user does not alter the defaults, select the first device capable of supporting the given precision, as presented in the table below.

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 4 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/auto_device_selection.md:121]: I guess w should, since this is the new approach. In the future, we need to work on organizing these files, though. Thanks @tiger100256-hu for dealing with snippets here.

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 5 above)
Comment (kblaszczak-intel) [reply docs/OV_Runtime_UG/auto_device_selection.md:207]: I have put my proposition with the link in a comment below. In this case we can remove the code.

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 8 above)
[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply docs/OV_Runtime_UG/auto_device_selection.md:16]: we don't have precision for model, precision is valid only for tensors in the model. Model is just a set of operations computing some tensors.

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 12]
```diff
@@ -13,30 +13,27 @@
...
++----------+------------------------------------------------------+-------------------------------------+
+| 3        | | Intel® Movidius™ Myriad™ X VPU                     | FP16                                |
+|          | | (e.g. Intel® Neural Compute Stick 2 (Intel® NCS2)) |                                     |
++----------+------------------------------------------------------+-------------------------------------+
+| 4        | | Intel® CPU                                         | FP32, FP16, INT8, BIN               |
+|          | | (e.g. Intel® Core™ i7-1165G7)                      |                                     |
++----------+------------------------------------------------------+-------------------------------------+
 @endsphinxdirective
 
 To put it simply, when loading the network to the first device on the list fails, AUTO will try to load it to the next device in line, until one of them succeeds. For example:
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:39]: can we provide a picture from the blob about auto plugin? it's hard to read the text w/o any illustration

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 13]
```diff
@@ -74,27 +71,27 @@ Inference with AUTO is configured similarly to other plugins: first you configur
 Following the OpenVINO™ naming convention, the Auto-Device plugin is assigned the label of “AUTO.” It may be defined with no additional parameters, resulting in defaults being used, or configured further with the following setup options:
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:71]: the text above provide performance numbers, @myshevts is it valid to provide any performance in docs? I suppose some approvals are needed here

Secondly, can we move BA usage example to the end as for other plugins? In the intro we need to provide only information about what plugin does, no more

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 14]
```diff
@@ -105,7 +102,6 @@ Following the OpenVINO™ naming convention, the Auto-Device plugin is assigned
    - replace {ov::device:priorities, "GPU,CPU"} with {"MULTI_DEVICE_PRIORITIES", "GPU,CPU"}
    - replace {ov::hint:model_priority, "LOW"} with {"MODEL_PRIORITY", "LOW"}
    - InferenceEngine::MultiDeviceConfigParams::KEY_MULTI_DEVICE_PRIORITIES is defined as same string "MULTI_DEVICE_PRIORITIES"
-   - CommonTestUtils::DEVICE_GPU + std::string(",") + CommonTestUtils::DEVICE_CPU is equal to "GPU,CPU"
    - InferenceEngine::PluginConfigParams::KEY_MODEL_PRIORITY is defined as same string "MODEL_PRIORITY"
    - InferenceEngine::PluginConfigParams::MODEL_PRIORITY_LOW is defined as same string "LOW"
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:106]: why do we need all these migration steps at all?

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 4 above)
Comment (ilya-lavrenov) [reply docs/OV_Runtime_UG/auto_device_selection.md:121]: do we need to demonstrate old API 1.0? In other documents we use only API 2.0

File: docs/OV_Runtime_UG/auto_device_selection.md [hunk 15]
```diff
@@ -234,7 +224,7 @@ For Python API
 
 
 ### Performance Hints
```
Comment (ilya-lavrenov) [inline docs/OV_Runtime_UG/auto_device_selection.md:226]: do we need such big section about performance hints? I don't see anything specific for AUTO here. Can we just refer to common document about per hints written by @myshevts  ?

File: docs/OV_Runtime_UG/auto_device_selection.md (hunk 13 above)
Comment (myshevts) [reply docs/OV_Runtime_UG/auto_device_selection.md:71]: I suggest to the the image (load-inference over timne) from the blog here

Comment (myshevts) [reply docs/OV_Runtime_UG/auto_device_selection.md:71]: this would obviate the need for the perf numbers

**Review by ilya-lavrenov — APPROVED** [review]
Please, fix the table generation and after we can merge the PR.

**Comment by ilya-lavrenov** [comment]
@tiger100256-hu please, port to the releases/2022/1 branch
